shell/src/main/java/org/crsh/console/Editor.java

```diff
@@ -115,7 +115,7 @@ void addToHistory(String line) {
    * @return the current bound
    */
   int getCursorBound() {
-    if (console.getMode() == Mode.EMACS) {
+    if (Mode.EMACS.equals(console.getMode())) {
```

Comment:
The Mode instances are static and final.  Object.equals would do the same comparison, and this check for equality takes away from the simplicity of the line.  What benefit does it provide?  See more detail in my first comment in this file.